cmd/simulator/txs/tx_generator.go

```diff
@@ -7,36 +7,46 @@ import (
 	"context"
 	"crypto/ecdsa"
 	"fmt"
+	"time"
 
 	"github.com/ava-labs/subnet-evm/core/types"
 	"github.com/ava-labs/subnet-evm/ethclient"
 	ethcrypto "github.com/ethereum/go-ethereum/crypto"
 )
 
-var _ TxSequence[*types.Transaction] = (*txSequence)(nil)
+var _ TxSequence[TimeTx] = (*txSequence)(nil)
 
 type CreateTx func(key *ecdsa.PrivateKey, nonce uint64) (*types.Transaction, error)
 
+type TimeTx struct {
+	Tx                             *types.Transaction
+	IssuanceDuration               time.Duration
+	ConfirmationDuration           time.Duration
+	IssuanceStart                  time.Time
+	IssuanceToConfirmationDuration time.Duration
```

Comment:
if this is equal to `IssuanceDuration + ConfirmationDuration` doesn't seem like we need to track it explicitly

Reply:
yeah i can probably remove this and simplify.